crates/paymaster-execution/src/tokens/mod.rs

```diff
@@ -91,17 +79,16 @@ impl TokenService {
     }
 
     /// Creates a new token service based on chain ID.
-    pub fn for_chain(chain_id: Felt) -> Self {
-        if chain_id == CHAIN_ID_SEPOLIA {
+    pub fn new(chain_id: ChainID) -> Self {
+        if chain_id.as_felt() == ChainID::Sepolia.as_felt() {
```

Comment:
Here I think we can have a "match" instead on the ChainID and not the felt. That way we don't have the "else" -> mainnet but each case is properly treated and if someone add something in ChainID, it will know that he has to update this match as well